identity-service/src/routes/protocol.js

```diff
@@ -0,0 +1,80 @@
+const express = require('express')
+const { recoverPersonalSignature } = require('eth-sig-util')
+
+const {
+  handleResponse,
+  successResponse,
+  errorResponseBadRequest,
+  errorResponse: formatResponse,
+  sendResponse
+} = require("../apiHelpers")
+const models = require("../models")
+
+const protocolRouter = express.Router()
+
+protocolRouter.get('/:wallet/delegation/minimum', handleResponse(async (req, res, next) => {
+  const { wallet } = req.params
+
+  const serviceProvider = await models.ProtocolServiceProviders.findOne({
+    where: { wallet: wallet.toLowerCase() }
+  })
+
+  if (serviceProvider === null) {
+    return formatResponse(404, 'Minimum delegation amount not set')
+  }
+
+  return successResponse({
+    wallet,
+    minimumDelegationAmount: serviceProvider.minimumDelegationAmount
+  })
+}))
+
+async function serviceProviderAuthMiddleware (req, res, next) {
+  try {
+    const encodedDataMessage = req.get('Encoded-Data-Message')
+    const signature = req.get('Encoded-Data-Signature')
+
+    if (!encodedDataMessage) throw new Error('[Error]: Encoded data missing')
+    if (!signature) throw new Error('[Error]: Encoded data signature missing')
+
+    const wallet = recoverPersonalSignature({ data: encodedDataMessage, sig: signature })
+    req.authedWallet = wallet
+    next()
+  } catch (err) {
+    req.logger.warn(`${err}`)
+    const errorResponse = errorResponseBadRequest('[Error]: ')
+    return sendResponse(req, res, errorResponse)
+  }
+}
+
+protocolRouter.post('/:wallet/delegation/minimum', serviceProviderAuthMiddleware, handleResponse(async (req, res, next) => {
+  const { wallet } = req.params
```

Comment:
technically you don't need to pass this param right?

Reply:
yeah, would just feel weird w/ the way the endpoint was structured though. thoughts on that?

Reply:
yeah this is fine